// certstream/lib/browser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _reconnectingwebsocket = require("reconnectingwebsocket");

var _reconnectingwebsocket2 = _interopRequireDefault(_reconnectingwebsocket);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var CertStreamClient = function () {
    function CertStreamClient(callback) {
        var skipHeartbeats = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

        _classCallCheck(this, CertStreamClient);

        this.context = {};
        this.callback = callback;
        this.skipHeartbeats = skipHeartbeats;
    }

    _createClass(CertStreamClient, [{
        key: "connect",
        value: function connect() {
            var _this = this;

            console.log("Connecting...");

            this.ws = new _reconnectingwebsocket2.default("wss://certstream.calidog.io/");

            console.log("Created ws -> ", this.ws);

            this.ws.onmessage = function (message) {
                console.log("onmessage called!");
                var parsedMessage = JSON.parse(message.data);

                if (parsedMessage.message_type === "heartbeat" && _this.skipHeartbeats) {
                    return;
                }

                _this.callback(message, _this.context);
            };

            this.ws.onopen = function () {
                console.log("Connection established to certstream! Waiting for messages...");
            };

            this.ws.open();
        }
    }]);

    return CertStreamClient;
}();

exports.default = CertStreamClient;
;
module.exports = exports["default"];
